### ISSUE TITLE
Django not working properly after v1.7

### ISSUE THREAD

**Issue by Tragio** [emmett-framework/granian#469]
Hi there 👋 

After version 1.7 (including 1.7.1), my Django is not working properly. If I run the following command:
```
uv run granian main.asgi:application \
    --host 0.0.0.0 \
    --port 8000 \
    --interface asginl \
    --access-log
```
I get this (relates to Django Channels):
```
TypeError: ProtocolTypeRouter.__call__() missing 1 required positional argument: 'send'
[ERROR] Application callable raised an exception
Traceback (most recent call last):
  File "/code/.venv/lib/python3.12/site-packages/granian/_futures.py", line 11, in future_watcher
    await inner(watcher.scope, watcher.proto)
  File "/code/.venv/lib/python3.12/site-packages/granian/rsgi.py", line 81, in _http_logger
    rv = await callback(scope, proto)
```
---
If I add the `task-impl` I get a weirder error, but I can't understand where this is coming from as it doesn't seem to be related to the WebSocket:
```
uv run granian main.asgi:application \
    --host 0.0.0.0 \
    --port 8000 \
    --interface asginl \
    --task-impl asyncio \
    --access-log
```
And this is what I get:
```
[INFO] Starting granian (main PID: 166)
[INFO] Listening at: http://0.0.0.0:8000
[INFO] Spawning worker-1 with pid: 167
[INFO] Started worker-1
[INFO] Started worker-1 runtime-1
2024-12-21T13:19:56.369374Z [info     ] request_started                [django_structlog.middlewares.request] ...
2024-12-21T13:19:56.643088Z [info     ] request_finished               [django_structlog.middlewares.request] code=200 ...
thread '<unnamed>' panicked at /rustc/90b35a6239c3d8bdabc530a6a0816f7ff89a0aaf/library/core/src/ptr/mod.rs:574:1:
Cannot drop pointer into Python heap without the GIL being held.
stack backtrace:
   0:     0xe5efe424ad34 - <unknown>
   1:     0xe5efe418a380 - <unknown>
   2:     0xe5efe4224808 - <unknown>
   3:     0xe5efe424bac0 - <unknown>
   4:     0xe5efe424b944 - <unknown>
   5:     0xe5efe424bfd8 - <unknown>
   6:     0xe5efe424bb58 - <unknown>
   7:     0xe5efe424baf8 - <unknown>
   8:     0xe5efe424baec - <unknown>
   9:     0xe5efe3e351d0 - <unknown>
  10:     0xe5efe3e8703c - <unknown>
  11:     0xe5efe3e79dd4 - <unknown>
  12:     0xe5efe3e87910 - <unknown>
  13:     0xe5efe424c6d4 - <unknown>
  14:     0xe5efe4edee90 - <unknown>
  15:     0xe5efe4f47b1c - <unknown>
  16:                0x0 - <unknown>
[ERROR] Unexpected exit from worker-1
[INFO] Shutting down granian
```

Thank you very much for the awesome work 😄

**Comment by gi0baro** [comment]
@Tragio I'm not sure what's going on, but from the your traceback

```
File "/code/.venv/lib/python3.12/site-packages/granian/rsgi.py", line 81, in _http_logger
    rv = await callback(scope, proto)
```

it seems is using the RSGI interface in place of ASGI.

**Comment by JasonLovesDoggo** [comment]
I assume it's what Giovanni said as I'm running on v1.7.1 and django 4.2 with granian & django working perfectly

**Comment by Tragio** [comment]
Thank you @gi0baro and @JasonLovesDoggo for your time. 

It is hard to understand the root cause as everything will work fine if I downgrade to 1.6.4, in my case I'm using the latest version of Django.

```
uv run granian main.asgi:application \
    --host 0.0.0.0 \
    --port 8000 \
    --interface asginl \
    --access-log
```

My asgi file:

```py
import os

from channels.auth import AuthMiddlewareStack
from channels.routing import ProtocolTypeRouter, URLRouter
from channels.security.websocket import AllowedHostsOriginValidator
from app.consumers import StatusConsumer
from django.core.asgi import get_asgi_application
from django.urls import re_path

os.environ.setdefault("DJANGO_SETTINGS_MODULE", "main.settings")

application = ProtocolTypeRouter(
    {
        "http": get_asgi_application(),
        "websocket": AllowedHostsOriginValidator(
            AuthMiddlewareStack(
                URLRouter([re_path(r"^ws/status/$", StatusConsumer.as_asgi())])
            )
        ),
    }
)
```

**Comment by Tragio** [comment]
@gi0baro I created a minimal repo: https://github.com/Tragio/django-granian-repro

You need to do `docker build -t myapp .` and `docker run -p 8000:8000 myapp` and you see the error coming. 🤔

**Comment by gi0baro** [comment]
> @gi0baro I created a minimal repo: https://github.com/Tragio/django-granian-repro

You're missing the `--interface asginl` part in `CMD` line at https://github.com/Tragio/django-granian-repro/blob/main/Dockerfile#L40

**Comment by Tragio** [comment]
> > @gi0baro I created a minimal repo: [Tragio/django-granian-repro](https://github.com/Tragio/django-granian-repro)
> 
> You're missing the `--interface asginl` part in `CMD` line at [Tragio/django-granian-repro@`main`/Dockerfile#L40](https://github.com/Tragio/django-granian-repro/blob/main/Dockerfile#L40)

Ahhh I forgot to add it to the `Dockerfile`, I already added.

```sh
❯ docker run -p 8000:8000 myapp
[INFO] Starting granian (main PID: 9)
[INFO] Listening at: http://0.0.0.0:8000
[INFO] Spawning worker-1 with pid: 10
[INFO] Started worker-1
[INFO] Started worker-1 runtime-1
thread '<unnamed>' panicked at /rustc/90b35a6239c3d8bdabc530a6a0816f7ff89a0aaf/library/core/src/ptr/mod.rs:574:1:
Cannot drop pointer into Python heap without the GIL being held.
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[ERROR] Unexpected exit from worker-1
[INFO] Shutting down granian
```

**Comment by gi0baro** [comment]
@Tragio the panic issue should be fixed in the upcoming 1.7.2
